cmd/state-installer/installer_windows.go

```diff
@@ -84,3 +72,67 @@ func (i *Installer) PrepareBinTargets(useBinDir bool) error {
 
 	return nil
 }
+
+func (i *Installer) cleanInstallPath() error {
+	if fileutils.DirExists(i.path) {
+		files, err := ioutil.ReadDir(i.path)
+		if err != nil {
+			return errs.Wrap(err, "Could not installation directory: %s", i.path)
+		}
+
+		for _, file := range files {
+			fname := strings.ToLower(file.Name())
+			targetFile := filepath.Join(i.path, file.Name())
+			if isStateExecutable(fname) {
+				renamedFile := filepath.Join(i.path, fmt.Sprintf("%s-%d.old", fname, time.Now().Unix()))
+				if err := os.Rename(targetFile, renamedFile); err != nil {
+					return errs.Wrap(err, "Could not rename corrupted executable: %s to %s", targetFile, renamedFile)
+				}
+				// This will likely fail but we try anyways
+				os.Remove(renamedFile)
+			}
+		}
+	}
+
+	isAdmin, err := osutils.IsAdmin()
+	if err != nil {
+		return errs.Wrap(err, "Could not determine if running as Windows administrator")
+	}
+
+	// Since we are repairing a corrupted install we need to also remove the old
+	// PATH entry. The new PATH entry will be added later in the install/update process.
+	// This is only an issue on Windows as on other platforms we can simply rewrite
+	// the PATH entry.
+	s := subshell.New(i.cfg)
+	if err := s.CleanUserEnv(i.cfg, sscommon.InstallID, !isAdmin); err != nil {
```

Comment:
This should use the `isAdmin` value that we record at install time (ie. if the user installed State Tool as admin then we should also try to remove env vars here as admin, and if they aren't running as admin then we error out).

We should probably do this check WAY at the start of the updater. Meaning if you installed as admin then you HAVE to update as admin.